src/models/setting.model.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  BelongsTo,
+  Column,
+  Model,
+  Table,
+  DeletedAt,
+  CreatedAt,
+  UpdatedAt,
+  ForeignKey,
+  AllowNull,
+  PrimaryKey,
+  DataType,
+} from 'sequelize-typescript';
+import Account from 'src/models/account.model';
+
+@Table({
+  indexes: [{ fields: ['account_id', 'name'], unique: true }],
+})
+export default class Setting extends Model {
+  @PrimaryKey
+  @Column(DataType.INTEGER)
+  id: number;
+
+  @AllowNull(false)
+  @Column(DataType.TEXT)
+  name: string;
+
+  @AllowNull(false)
+  @Column({ type: DataType.ENUM('number', 'string', 'object', 'boolean') })
+  data_type: 'number' | 'string' | 'object' | 'boolean';
```

Comment:
Please create a typescript string enum instead of a uninon of strings
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums
These are much better in readability and we can use this in DTOs, as a type and even inside DataType.ENUM.